src/track/beatutils.cpp

```diff
@@ -254,52 +247,69 @@ QVector<double> BeatUtils::calculateIronedBeatMap(
     double previousPhaseErrorMean = 0.0;
     auto phaseErrorAvarageCalculator = MovingAvarage(1);
     auto bpmAvarageCalculator = MovingAvarage(1);
-    double meanBpm;
+    double meanBpm = 0.0;
+    double meanBeatLength = 0.0;
+    bool foundFistOutlier = false;
     QVector<double> fixedBeats;
-    while (rightIndex < rawbeats.size()) {
+    while (beatOffset < rawbeats.last() + meanBeatLength) {
         double bpm = 60.0 * sampleRate / (rawbeats[rightIndex] - rawbeats[rightIndex -1]);
         meanBpm = bpmAvarageCalculator(bpm);
```

Comment:
can we avoid the bpm round trip here, and use the beat length instead?

Reply:
We even can omit the  use of bpmAvarageCalculator. 
Just 
```
(rawbeats[rightIndex] - rawbeats[leftIndex]) /  (rightIndex - leftIndex) 
```

Reply:
Awesome, done!